### ISSUE TITLE
💅 `noFallthroughSwitchClause` false positive for nested exhaustive `switch`/`case`

### ISSUE THREAD

**Issue by lensbart** [biomejs/biome#5379]
### Environment information

```bash
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.13.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  GraphQL enabled:              false
  Recommended:                  false
  All:                          false
  Enabled rules:
  (…)
  suspicious/noFallthroughSwitchClause
  (…)

Workspace:
  Open Documents:               0
```

### Rule name

noFallthroughSwitchClause

### Playground link

https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABFAHgAYQBtAHAAbABlACgAewAKAAkAbABlAHQAdABlAHIALAAKAAkAbgB1AG0AYgBlAHIALAAKAH0AOgAgAHsAIABsAGUAdAB0AGUAcgA6ACAAIgBhACIAIAB8ACAAIgBiACIAIAB8ACAAIgBjACIAOwAgAG4AdQBtAGIAZQByADoAIAAxACAAfAAgADIAIAB9ACkAIAB7AAoACQBzAHcAaQB0AGMAaAAgACgAbABlAHQAdABlAHIAKQAgAHsACgAJAAkAYwBhAHMAZQAgACIAYQAiADoACgAJAAkACQByAGUAdAB1AHIAbgAgACIAYQAiAAoACQAJAGMAYQBzAGUAIAAiAGIAIgA6ACAAewAKAAkACQAJAHMAdwBpAHQAYwBoACAAKABuAHUAbQBiAGUAcgApACAAewAKAAkACQAJAAkAYwBhAHMAZQAgADEAOgAKAAkACQAJAAkACQByAGUAdAB1AHIAbgAgADEACgAJAAkACQAJAGMAYQBzAGUAIAAyADoACgAJAAkACQAJAAkAcgBlAHQAdQByAG4AIAAyAAoACQAJAAkAfQAKAAkACQB9AAoACQAJAGMAYQBzAGUAIAAiAGMAIgA6AAoACQAJAAkAcgBlAHQAdQByAG4AIAAiAGMAIgAKAAkAfQAKAH0A

### Expected result

No lint error

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
I'm inclined to say this is intended behavior because technically, typescript types are not runtime assertions. We're also very limited on what we can do with type information right now (see #3187 ).

Does eslint also produce this false positive?

**Comment by ematipico** [comment]
[Eslint](https://eslint.org/play/#eyJ0ZXh0IjoiZXhwb3J0IGZ1bmN0aW9uIEV4YW1wbGUoe1xuXHRsZXR0ZXIsXG5cdG51bWJlcixcbn0pIHtcblx0c3dpdGNoIChsZXR0ZXIpIHtcblx0XHRjYXNlIFwiYVwiOlxuXHRcdFx0cmV0dXJuIFwiYVwiXG5cdFx0Y2FzZSBcImJcIjoge1xuXHRcdFx0c3dpdGNoIChudW1iZXIpIHtcblx0XHRcdFx0Y2FzZSAxOlxuXHRcdFx0XHRcdHJldHVybiAxXG5cdFx0XHRcdGNhc2UgMjpcblx0XHRcdFx0XHRyZXR1cm4gMlxuXHRcdFx0fVxuXHRcdH1cblx0XHRjYXNlIFwiY1wiOlxuXHRcdFx0cmV0dXJuIFwiY1wiXG5cdH1cbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1iaW5hcnktZXhwcmVzc2lvbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1lbXB0eS1zdGF0aWMtYmxvY2siOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbmV3LW5hdGl2ZS1ub25jb25zdHJ1Y3RvciI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11bnVzZWQtcHJpdmF0ZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLXVudXNlZC12YXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl0sIm5vLWNvbnNvbGUiOlsiZXJyb3IiXSwibm8tYWxlcnQiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19) matches the behaviour.

Biome doesn't infer types yet, so I think we can close it for now and maybe enhance it once we have the infrastructure.
